src/toolkit/matl_buy_policy.cc

```diff
@@ -55,34 +53,46 @@ void MatlBuyPolicy::set_throughput(double x) {
   throughput_ = x;
 }
 
-void MatlBuyPolicy::set_active(int x) {
-  assert(x > 0);
-  active_ = x;
+void MatlBuyPolicy::set_default_distributions() {
+  active_dist_ = new FixedIntDist(1);
+  dormant_dist_ = new FixedIntDist(-1);
+  size_dist_ = new FixedDoubleDist(1.0);
 }
 
-void MatlBuyPolicy::set_dormant(int x) {
-  assert(x >= 0);
-  dormant_ = x;
+void MatlBuyPolicy::init_active_dormant() {
+  next_active_end_ = active_dist_->sample();
+ 
+  int dormant_len = dormant_dist_->sample();
+  if (dormant_len < 0) {
+    next_dormant_end_ = -1;
+  }
+  else {
+    next_dormant_end_ = dormant_len + next_active_end_;
+  }
```

Comment:
Assuming it makes sense to call `SetNextActiveEnd()` above, it will make this more readable if you just call `SetNextDormandEnd()` here as well.  As long as you set `next_dormant_end_` to be -1 before then, when appropriate, it will result in the same behavior, but waste a single random number.